File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/util/ExceptionUtils.java
package org.javamaster.invocationlab.admin.util;


import java.util.Arrays;
import java.util.stream.Collectors;

/**
 * @author yudong
 * @date 2023-12-30
 */
public class ExceptionUtils {

    public static String getSimplifyStackTrace(Throwable t) {
        StackTraceElement[] stackTrace = t.getStackTrace();
        String trace = Arrays.stream(stackTrace)
                .filter(it -> it.getLineNumber() != -1 && it.getClassName().startsWith("org.javamaster"))
                .limit(5)
                .map(it -> "\tat " + it)
                .collect(Collectors.joining("\r\n"));
        String error = t + "\r\n" + trace;
        if (t.getCause() != null) {
            String causeError = org.apache.commons.lang3.exception.ExceptionUtils.getStackTrace(t.getCause());
            error = error + "\r\nCaused by:" + causeError;
        }
        return error;
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/util/ExceptionUtils.kt
package org.javamaster.invocationlab.admin.util

import org.apache.commons.lang3.exception.ExceptionUtils
import java.util.*
import java.util.stream.Collectors


/**
 * @author yudong
 * @date 2023-12-30
 */
object ExceptionUtils {
    @JvmStatic
    fun getSimplifyStackTrace(t: Throwable): String {
        val stackTrace = t.stackTrace
        val trace = Arrays.stream(stackTrace)
            .filter { it: StackTraceElement -> it.lineNumber != -1 && it.className.startsWith("org.javamaster") }
            .limit(5)
            .map { it: StackTraceElement -> "\tat $it" }
            .collect(Collectors.joining("\r\n"))
        var error = """
             $t
             $trace
             """.trimIndent()
        if (t.cause != null) {
            val causeError = ExceptionUtils.getStackTrace(t.cause)
            error = "$error\r\nCaused by:$causeError"
        }
        return error
    }
}
